kubescaler/scaler/aws/cluster.py

```diff
@@ -588,25 +798,39 @@ def delete_cluster(self):
         And let's go backwards - deleting first what we created last.
         """
         logger.info("🔨️ Deleting node workers...")
-        self.delete_workers_stack()
+        if self.eks_nodegroup:
+            self.delete_nodegroup(self.node_group_name)
+        else:
+            self.delete_workers_stack()
         # We could delete keypair, but let's keep for now, assuming could be reused elsewhere
         # and a deletion might be unexpected to the user
 
-        # Now delete the cluster
-        try:
-            self.eks.delete_cluster(name=self.cluster_name)
-        except Exception as e:
-            logger.info(f"⏳️ Cluster likely already deleted: {e}")
-            return
-
+        # An error occurred (ResourceInUseException) when calling the DeleteCluster operation: Cannot delete because cluster <name> currently has an update in progress
+        # TODO Make a good design for this portion
+        while True:
+            try:
+                self.eks.delete_cluster(name=self.cluster_name)
+            except self.eks.exceptions.ResourceInUseException as e:
+                print(f"The cluster resources are busy, retry in a few seconds")
+                time.sleep(60)
+                continue
+            except self.eks.exceptions.ResourceNotFoundException as e:
+                print(f"⏳️ Cluster likely already deleted: {e}")
+                logger.info(f"⏳️ Cluster likely already deleted: {e}")
+                return
+            break
+        print("⏳️ Cluster deletion started! Waiting...")
         logger.info("⏳️ Cluster deletion started! Waiting...")
```

Comment:
Same here - feel free to delete the logger.info. As an alternative, change the default logging level (or run setup_logger in this file) and then use it.